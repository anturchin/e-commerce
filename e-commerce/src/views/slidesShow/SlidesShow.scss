.slideshow-container {
    position: relative;
    width: 485px;
    padding: 40px;
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 1000;
    background: #222;
    border-radius: 4px;
    box-shadow: 0 15px 35px rgb(0, 0, 0);
    margin: 3rem 3rem;
}

.slides-wrapper {
    padding: 4rem 0;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.dot {
    cursor: pointer;
    height: 15px;
    width: 15px;
    margin: 0 2px;
    background-color: #bbb;
    border-radius: 50%;
    display: inline-block;
    transition: background-color 0.6s ease;
}

.dot-container {
    padding: 1rem 0 0 0;
}

.dot.active,
.dot:hover {
    background-color: #717171;
}

@media (max-width: 1500px) {
    .slideshow-container {
        width: 485px;
        margin: 2.5rem 2.5rem; /* Уменьшаем margin */
    }
}

@media (max-width: 1400px) {
    .slideshow-container {
        width: 450px;
        margin: 2rem 2rem;
    }
}

@media (max-width: 1300px) {
    .slideshow-container {
        width: 420px;
        margin: 1.5rem 1.5rem;
    }
}

@media (max-width: 1200px) {
    .slideshow-container {
        width: 390px;
        margin: 1rem 1rem;
    }
}

@media (max-width: 1100px) {
    .slideshow-container {
        width: 360px;
        margin: 0.5rem 0.5rem;
        padding: 35px;
    }
}

@media (max-width: 1000px) {
    .slideshow-container {
        width: 330px;
        margin: 0.5rem 0.5rem;
        padding: 35px;
    }
}

@media (max-width: 950px) {
    .slideshow-container {
        width: 315px;
        margin: 0.5rem 0.5rem;
        padding: 32px;
    }
}

@media (max-width: 900px) {
    .slideshow-container {
        width: 305px;
        margin: 0.5rem 0.5rem;
        padding: 30px;
    }
}

@media (max-width: 850px) {
    .slideshow-container {
        width: 295px;
        margin: 0.5rem 0.5rem;
        padding: 28px;
    }
}

@media (max-width: 800px) {
    .slideshow-container {
        width: 285px;
        margin: 0.3rem 0.3rem;
        padding: 26px;
    }
}

@media (max-width: 750px) {
    .slideshow-container {
        width: 280px;
        margin: 0.3rem 0.3rem;
        padding: 25px;
    }
}

@media (max-width: 700px) {
    .slideshow-container {
        width: 270px;
        margin: 0.2rem 0.2rem;
        padding: 25px;
    }
}

@media (max-width: 650px) {
    .slideshow-container {
        width: 265px;
        margin: 0.2rem 0.2rem;
        padding: 24px;
    }
}

@media (max-width: 600px) {
    .slideshow-container {
        width: 255px;
        margin: 0.1rem 0.1rem;
        padding: 22px;
    }
}

@media (max-width: 550px) {
    .slideshow-container {
        width: 220px;
        height: 150px;
        margin: 0.1rem 0.1rem;
        padding: 22px;
    }

    .dot {
        width: 12px;
        height: 12px;
    }
}

@media (max-width: 500px) {
    .slideshow-container {
        width: 210px;
        margin: 0.1rem 0.1rem;
    }
}

@media (max-width: 450px) {
    .slideshow-container {
        width: 200px;
        margin: 0.1rem 0.1rem;
    }
}

@media (max-width: 400px) {
    .slideshow-container {
        width: 190px;
        margin: 0.1rem 0.1rem;
    }
}

@media (max-width: 360px) {
    .slideshow-container {
        width: 180px;
        padding: 20px;
        margin: 0.5rem 0.5rem;
    }
}
