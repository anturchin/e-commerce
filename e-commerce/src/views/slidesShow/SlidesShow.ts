import { View } from '../View';
import { SliesItem } from './slidesItem/SlidesItem';
import { LinkPrev } from './linkPrev/LinkPrev';
import { LinkNext } from './linkNext/LinkNext';

import './SlidesShow.scss';

export class SlidesShow extends View {
    private items: SliesItem[] = [];

    private dots: HTMLElement[] = [];

    private slideIndex: number = 1;

    constructor(urls: string[]) {
        super({ tag: 'div', classNames: ['slideshow-container'] });
        this.setupSlidesShow(urls);
        this.showSlides(this.slideIndex);
    }

    private setupSlidesShow(urls: string[]): void {
        const slidesWrapper = document.createElement('div');
        slidesWrapper.classList.add('slides-wrapper');

        this.items = urls.map((url) => new SliesItem(url));

        const prevBtn = new LinkPrev().getElement();
        const nextBtn = new LinkNext().getElement();

        prevBtn.addEventListener('click', () => this.plusSlides(-1));
        nextBtn.addEventListener('click', () => this.plusSlides(1));

        this.getElement().append(prevBtn, slidesWrapper, nextBtn);

        const dotContainer = document.createElement('div');
        dotContainer.classList.add('dot-container');

        this.dots = this.items.map((_, index) => {
            const dot = document.createElement('span');
            dot.classList.add('dot');
            dot.addEventListener('click', () => this.currentSlide(index + 1));
            dotContainer.append(dot);
            return dot;
        });
        slidesWrapper.append(...this.items.map((item) => item.getElement()));
        slidesWrapper.append(dotContainer);
    }

    plusSlides(n: number) {
        this.showSlides((this.slideIndex += n));
    }

    currentSlide(n: number) {
        this.showSlides((this.slideIndex = n));
    }

    showSlides(n: number) {
        if (n > this.items.length) {
            this.slideIndex = 1;
        }
        if (n < 1) {
            this.slideIndex = this.items.length;
        }
        this.items.forEach((item, i) => {
            const element = item.getElement();
            element.style.display = 'none';
            this.dots[i].classList.remove('active');
        });
        this.items[this.slideIndex - 1].getElement().style.display = 'block';
        this.dots[this.slideIndex - 1].classList.add('active');
    }
}
