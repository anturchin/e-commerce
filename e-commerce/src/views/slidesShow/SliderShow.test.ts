import { describe, expect, test } from '@jest/globals';
import { SlidesShow } from './SlidesShow';

describe('SlidesShow', () => {
    test('should match', () => {
        const slidesShow = new SlidesShow(['url1', 'url2']);
        const renderedSlidesShow = slidesShow.getElement();
        expect(renderedSlidesShow.outerHTML).toMatchSnapshot();
    });
});
