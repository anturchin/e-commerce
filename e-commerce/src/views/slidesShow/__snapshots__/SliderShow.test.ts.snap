// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlidesShow should match 1`] = `"<div class="mySlides fade"><div class="slides" style="display: block;"><img class="img-slide modal-slide" src="url1"></div><div class="slides" style="display: none;"><img class="img-slide modal-slide" src="url2"></div><button class="prev">prev</button><button class="next">next</button></div>"`;
